llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -8666,6 +8717,29 @@ void AArch64TargetLowering::AdjustInstrPostInstrSelection(MachineInstr &MI,
     }
   }
 
+  if (MI.getOpcode() == AArch64::FORM_TRANSPOSED_REG_TUPLE_X2_PSEUDO ||
+      MI.getOpcode() == AArch64::FORM_TRANSPOSED_REG_TUPLE_X4_PSEUDO) {
+    // If input values to the FORM_TRANSPOSED_REG_TUPLE pseudo aren't copies
+    // from a StridedOrContiguous class, fall back on REG_SEQUENCE node.
+    if (!shouldUseFormStridedPseudo(MI)) {
```

Comment:
nit: Maybe fold those conditions into one? or otherwise bail out early if `shouldUseFormStridedPseudo(MI) == true`, to avoid a level of indentation?